v2/yarpcthrift/codec.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpcthrift
+
+import (
+	"bytes"
+
+	"go.uber.org/yarpc/v2/yarpcerror"
+
+	"go.uber.org/thriftrw/protocol"
+	"go.uber.org/thriftrw/wire"
+	yarpc "go.uber.org/yarpc/v2"
+)
+
+type thriftCodec struct {
+	protocol   protocol.Protocol
+	enveloping bool
+
+	responder protocol.Responder
+}
+
+func newCodec(protocol protocol.Protocol, enveloping bool) *thriftCodec {
+	return &thriftCodec{
+		protocol:   protocol,
+		enveloping: enveloping,
+	}
+}
+
+// setResponder checks whether a responder has been set and sets it to one. An
```

Comment:
nit since this is internal documentation: we can break up this comment into a few lines, short description as the first block, and then more details after.
eg
```
// setResponder sets a responder and returns an error if one is already set.
//
// An instance of a thrift codec should not set a responder more than once; if
// it does, it may not encode a response properly. We use this instead of
// directly setting a responder to ensure that we only set it once.
```